<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>animation_workbench_base</class>
 <widget class="QDialog" name="animation_workbench_base">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>830</width>
    <height>866</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_3">
   <item row="1" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>-297</y>
        <width>800</width>
        <height>1105</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout_7">
       <item row="0" column="0">
        <widget class="QGroupBox" name="groupBox_4">
         <property name="title">
          <string>Animation Reference</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <widget class="QLabel" name="point_layer_label">
            <property name="text">
             <string>Point layer</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="point_layer_combo"/>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QLabel" name="point_layer_help">
            <property name="text">
             <string>Select a point layer to use as the animation reference. The animation will fly from point to point in the layer according to the settings below.</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QGroupBox" name="easings_group">
         <property name="title">
          <string>Easings</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="fly_easing_label">
            <property name="text">
             <string>Fly Easing</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="fly_easing_combo"/>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QLabel" name="fly_easing_help">
            <property name="text">
             <string>The fly easing will determine the motion characteristics of the camera on the Y axis as it flies across the scene.</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="zoom_easing_label">
            <property name="text">
             <string>Zoom Easing</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QComboBox" name="zoom_easing_combo"/>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QLabel" name="zoom_easing_help">
            <property name="text">
             <string>The zoom easing will affect the behaviour of the camera during zoom transitions.</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QGroupBox" name="render_mode_group">
         <property name="title">
          <string>Render Mode</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QRadioButton" name="radio_sphere">
            <property name="text">
             <string>Sphere</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QRadioButton" name="radio_planar">
            <property name="text">
             <string>Planar</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QRadioButton" name="radio_static">
            <property name="text">
             <string>Static</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="render_mode_help">
            <property name="text">
             <string>The render mode determines the behaviour and type of the animation. For 'Sphere' the coordinate reference system (CRS) will be manipulated to create a spinning globe effect. For 'Plane', the CRS will not be altered, but will pan and zoom to each point. For 'Static' the animation will not not pan / zoom the map.Â´</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QGroupBox" name="output_destination_group">
         <property name="title">
          <string>Output Destination</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="0">
           <widget class="QLabel" name="folder_label">
            <property name="text">
             <string>Folder</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="folder_edit"/>
          </item>
          <item row="0" column="2">
           <widget class="QToolButton" name="folder_button">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="3">
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>The output folder will be populated with all of the frames of the animation, and the GIF or MP4 as selected above.</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QGroupBox" name="animation_frames_group">
         <property name="title">
          <string>Animation Frames</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_8">
          <item row="0" column="0">
           <widget class="QLabel" name="point_frames_label">
            <property name="text">
             <string>Frames between points</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="point_frames_spin"/>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QLabel" name="point_frames_help">
            <property name="text">
             <string>This is the number of frames that will be used during animation of the dwell period at each point. Video generation is done at 30 frames per second, so a value of 30 here would result in a 1 second dwell time.</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="hover_frames_label">
            <property name="text">
             <string>Hover frames at each point</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="hover_frames_spin"/>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QLabel" name="hover_frames_help">
            <property name="text">
             <string>This is the number of frames that will be used during animation of the motion from one point to the next. Video generation is done at 30 frames per second, so a value of 30 here would result in a 1 second flight time between two consecutive points.</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QGroupBox" name="output_options_group">
         <property name="title">
          <string>Output Options</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <item row="0" column="0">
           <widget class="QRadioButton" name="radio_gif">
            <property name="text">
             <string>Animated GIF</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="output_options_help">
            <property name="text">
             <string>Select which output format you would like. Regardless of which you choose, a folder of images will be created, one image per frame. For the GIF export to work, you will need to have the ImageMagick 'convert'  application available on your system. For the MP4 option to work, you need to have the 'ffmpeg' application on your system.</string>
            </property>
            <property name="wordWrap">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QRadioButton" name="rad_movie">
            <property name="text">
             <string>Movie (MP4)</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>animation_workbench_base</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>animation_workbench_base</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
